src/inference/AdvancedSMC.jl

```diff
@@ -147,72 +186,44 @@ function step(model, spl::Sampler{<:PG}, vi::VarInfo)
     ## pick a particle to be retained.
     Ws, _ = weights(particles)
     indx = randcat(Ws)
-    push!(spl.info[:logevidence], particles.logE)
-
-    return particles[indx].vi, true
-end
-
-function sample(  model::Model,
-                  alg::PG;
-                  save_state=false,         # flag for state saving
-                  resume_from=nothing,      # chain to continue
-                  reuse_spl_n=0             # flag for spl re-using
-                )
-
-    spl = reuse_spl_n > 0 ?
-          resume_from.info[:spl] :
-          Sampler(alg)
-    if resume_from != nothing
-        spl.selector = resume_from.info[:spl].selector
-    end
-    @assert typeof(spl.alg) == typeof(alg) "[Turing] alg type mismatch; please use resume() to re-use spl"
-
-    n = reuse_spl_n > 0 ?
-        reuse_spl_n :
-        alg.n_iters
-    samples = Vector{Sample}()
-
-    ## custom resampling function for pgibbs
-    ## re-inserts reteined particle after each resampling step
-    time_total = zero(Float64)
-
-    vi = resume_from == nothing ?
-        empty!(VarInfo(model)) :
-        resume_from.info[:vi]
-
-    pm = nothing
-    PROGRESS[] && (spl.info[:progress] = ProgressMeter.Progress(n, 1, "[PG] Sampling...", 0))
+    push!(spl.state.logevidence, particles.logE)
 
-    for i = 1:n
-        time_elapsed = @elapsed vi, _ = step(model, spl, vi)
-        push!(samples, Sample(vi))
-        samples[i].value[:elapsed] = time_elapsed
+    # Extract the VarInfo from the retained particle.
+    params = tonamedtuple(spl.state.vi)
+    spl.state.vi = particles[indx].vi
+    lp = getlogp(spl.state.vi)
 
-        time_total += time_elapsed
+    # update the master vi.
+    return transition(params, lp, Ws[indx], particles.logE)
```

Comment:
In PG/SMC, the particle weights are reset to `1` after each resampling step. Since here we resampled particles in L188, the weight should be one, that is `transition(params, lp, 1, particles.logE)`.

Reply:
Should I even store the weight field then? I was intending for it to go to the end user, but if it's always one, I can remove it.

Reply:
I think It's safe to remove it here.

Reply:
UPDATE: it might be helpful to keep the weight field for importance samplers, where we don't do resampling in general.

Reply:
Okay. Currently `IS` just returns the default `Transition` object, which only contains an `lp` field. If we wanted to do something like this we could just make a new `Transition` object for the importance sampler to grab the weights. I can do that right now, but I'm not sure how we should calculate weights here.